src/ApplicationSampler.cpp

```diff
@@ -424,6 +424,13 @@ namespace geopm
 #ifdef GEOPM_DEBUG
         std::cout << "Info: <geopm> ApplicationSampler::sampler_cpu(): The Controller will run on logical CPU " << result << std::endl;
 #endif
+
+        if (result == m_num_cpu - 1) {
```

Comment:
This should be triggered based on

    if (!found_inactive_core)  

Potentially we could add another bool and print a different message if we are sharing a core with the application, but on an unused hyperthread (found_insactive_thread set after line 419)?  Even more information could be printed if we are running on core 0 (sharing a core with the OS).